dash-pipeline/SAI/src/dashsai.cpp

```diff
@@ -705,6 +752,40 @@ bool DashSai::removeFromTable(
     return retCode == grpc::StatusCode::OK;
 }
 
+bool DashSai::getFromTable(
+        _In_ sai_object_id_t id,
+        _Out_ std::shared_ptr<p4::v1::TableEntry> &entry)
+{
+    DASH_LOG_ENTER();
+
+    if (!m_apiInitialized)
+    {
+        DASH_LOG_ERROR("api not initialized");
+
+        return false;
+    }
+
+    MUTEX;
+
+    auto range = m_tableEntryMap.equal_range(id);
```

Comment:
emmmm, do we really need a multi-map here? each object id should be unique with the object id encoding, am I right? @kcudnik

Reply:
@r12f , `map` is enough. fix it at https://github.com/sonic-net/DASH/pull/653